### PR TITLE
Update MNIST Colab notebook and its output models.

### PR DESCRIPTION
**Pull request by ScottTodd**
As part of this, I'm moving the full model with its weights (800KB) to `iree/samples/models/` and removing the check lines from it. I'm keeping a version with dummy weights in `iree/test/e2e/models/` for test coverage. These files should still be updated by running the notebook, not by direct edits.

### CODE REVIEW

**Comment by MaheshRavishankar** [comment]
looks good to me.

File: iree/test/e2e/models/BUILD [hunk 1]
```diff
@@ -25,7 +25,6 @@ iree_lit_test_suite(
     name = "lit",
     srcs = glob(
         ["*.mlir"],
```
Comment (GMNGeoffrey) [inline iree/test/e2e/models/BUILD:27]: Nit: can you fold this all onto one line, now that it's got no excludes?

**Comment by ScottTodd** [comment]
> Thanks! Can you add a note that we should really figure out a way to not check this into source? The best note would be a link to an issue :-P

That's mentioned in the new README:

```
Note: Models placed here should be small enough to be stored comfortably in git.
If a model file has large weights, please store those outside of git and fetch
them on-demand.
```

(we don't actually have a way to do that yet though :P)

File: iree/test/e2e/models/BUILD (hunk 1 above)
Comment (ScottTodd) [reply iree/test/e2e/models/BUILD]: Sure.

**Pull request closed without merging**